### PR TITLE
[refactor] removed legacy SWATH chromatogram extraction code

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Description

For discussion:
while going through the ChromatogramExtractor code I think I stumbled over some (potentially) dead code.
I think the template is only used in pyopenms (at least there are some tests).
I think it might make sense to do this for other parts of OpenSWATH to better see what is used and how.

Even if we still need it in pyopenms, we could remove most code and just call the more modern version with ChromatogramExtractorAlgorithm.

TODO:

- update python example files to use ChromatogramExtractorAlgorithm

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

___

### **PR Type**
enhancement

___

### **Description**
- Removed legacy methods `extractChromatograms`, `extract_value_tophat`, and `extract_value_bartlett` from `ChromatogramExtractor.h`, simplifying the codebase.
- Enhanced documentation for the `ChromatogramExtractor` class to clarify its functionality and interfaces.
- Removed commented-out legacy code in `ChromatogramExtractor.pxd` to clean up the code.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>ChromatogramExtractor.pxd</strong><dd><code>Remove legacy commented-out code in ChromatogramExtractor.pxd</code></dd></summary>
<hr>

src/pyOpenMS/pxds/ChromatogramExtractor.pxd

<li>Removed commented-out legacy code related to <code>extract_value_tophat</code> and <br><code>extract_value_bartlett</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7658/files#diff-25d85dd89ef6bb1ec3000e9ac4542c63aded7111b3c79810ef375b479a85fe9a">+1/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ChromatogramExtractor.h</strong><dd><code>Refactor ChromatogramExtractor by removing legacy methods</code></dd></summary>
<hr>

src/openms/include/OpenMS/ANALYSIS/OPENSWATH/ChromatogramExtractor.h

<li>Updated class documentation for <code>ChromatogramExtractor</code>.<br> <li> Removed legacy method <code>extractChromatograms</code> with detailed <br>implementation.<br> <li> Removed legacy methods <code>extract_value_tophat</code> and <br><code>extract_value_bartlett</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7658/files#diff-84b74f39687f666e4ddf25c2bc7a327fa9ddb7f1ae82c98bf8be24b3610b73ee">+26/-283</a></td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: src/pyOpenMS/pxds/ChromatogramExtractor.pxd [hunk 1]
```diff
@@ -24,9 +24,5 @@ cdef extern from "<OpenMS/ANALYSIS/OPENSWATH/ChromatogramExtractor.h>" namespace
                                   double rt_extraction_window,
```
Comment (timosachsenberg) [inline src/pyOpenMS/pxds/ChromatogramExtractor.pxd]: I think this might be the only use of it. Important?

**Comment by timosachsenberg** [comment]
@jcharkow @singjc do we use this for e.g. plotting or other downstream scripts?

**Comment by jcharkow** [comment]
Thank you for looking into this. Yes there is definitely a lot of dead code with OpenSWATH. Another section which I will open in another PR is potentially removing the SONAR stuff. This was never used in practice and was only experimental.

**Comment by singjc** [comment]
We don't use the `ChromatogramExtractor` in pyopenms_viz or massdash, I don't think we use it in any other downstream scripts either. Not sure if maybe @alhigaylan uses this method in his scripts for his work though.

I agree, I think there is a lot of unused code in OpenSwath that could probably be cleaned to make things less confusing. As Josh said, I don't think the SONAR stuff is used, it was only added when we were initially investigating SONAR and timsTOF, but the latter seemed more promising. 

On the note of cleaning things up though, I have been working on incorporating ion mobility information into the IPF framework for PTMs, which sometimes makes it challenging for me when the OpenSwath related codebase changes, because it sometimes affects the version that I'm working on. I will open a PR for this soonish, but I first want to make sure the implementation has a useful gain for the end results. Which currently it seems to.

**Comment by jcharkow** [comment]
On the topic of OpenSWATH refactoring. I have opened a discussion. https://github.com/OpenMS/OpenMS/discussions/7665
Please let me know your thoughts.

**Comment by timosachsenberg** [comment]
> We don't use the `ChromatogramExtractor` in pyopenms_viz or massdash, I don't think we use it in any other downstream scripts either. Not sure if maybe @alhigaylan uses this method in his scripts for his work though.
> 
> I agree, I think there is a lot of unused code in OpenSwath that could probably be cleaned to make things less confusing. As Josh said, I don't think the SONAR stuff is used, it was only added when we were initially investigating SONAR and timsTOF, but the latter seemed more promising.
> 
> On the note of cleaning things up though, I have been working on incorporating ion mobility information into the IPF framework for PTMs, which sometimes makes it challenging for me when the OpenSwath related codebase changes, because it sometimes affects the version that I'm working on. I will open a PR for this soonish, but I first want to make sure the implementation has a useful gain for the end results. Which currently it seems to.

ok good to know. I will hold back with merging changes to OpenSWATH until the IPF part is merged (or you think it doesn't affect anything.

**Comment by alhigaylan** [comment]
> We don't use the `ChromatogramExtractor` in pyopenms_viz or massdash, I don't think we use it in any other downstream scripts either. Not sure if maybe @alhigaylan uses this method in his scripts for his work though.
> 
> I agree, I think there is a lot of unused code in OpenSwath that could probably be cleaned to make things less confusing. As Josh said, I don't think the SONAR stuff is used, it was only added when we were initially investigating SONAR and timsTOF, but the latter seemed more promising.
> 
> On the note of cleaning things up though, I have been working on incorporating ion mobility information into the IPF framework for PTMs, which sometimes makes it challenging for me when the OpenSwath related codebase changes, because it sometimes affects the version that I'm working on. I will open a PR for this soonish, but I first want to make sure the implementation has a useful gain for the end results. Which currently it seems to.

 From my end, I do agree with removing dead code. I do not use `ChromatogramExtractor` in my scripts.

**Review by jcharkow — APPROVED** [review]
Looks good to me

**Comment by poshul** [comment]
Looks like this caused some issues with our pyopenms unit tests: https://github.com/OpenMS/OpenMS/actions/runs/11809391381/job/32899599513#step:10:182

**Comment by timosachsenberg** [comment]
thanks. will fix

**Pull request merged**